File: client/src/lib/types/timesheet-status-history.ts
export interface TimesheetStatusHistory {
    id: string
    timesheetId: string
    employeeId: string
    employeeName: string
    changedByUserId: string
    changedByUserName: string
    oldStatus: string | null
    newStatus: string
    comment: string | null
    changedAt: string
}

File: Domain/TimesheetStatusHistory.cs
using System;

namespace Domain;

public class TimesheetStatusHistory
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string TimesheetId { get; set; } = string.Empty;
    public Timesheet? Timesheet { get; set; }
    public string ChangedByUserId { get; set; } = string.Empty;
    public User? ChangedByUser { get; set; }
    public int FromStatus { get; set; }
    public int ToStatus { get; set; }
    public string? Comment { get; set; }
    public DateTime ChangedAt { get; set; } = DateTime.UtcNow;
}
